.github/workflows/test.yml

```diff
@@ -35,16 +45,32 @@ jobs:
       with:
         ruby-version: ${{ matrix.ruby }}
     - run: ruby --version
-    - run: ridk version
-      if: matrix.os == 'windows-latest' && !startsWith(matrix.ruby, 'jruby') && !startsWith(matrix.ruby, '2.3') && !startsWith(matrix.ruby, '2.2')
+    - name: gcc and ridk version
+      if: matrix.os == 'windows-latest'
+      run: |
+        # show build gcc version and run 'ridk version'
+        ruby -e "puts 'build gcc: ' + RbConfig::CONFIG.fetch('CC_VERSION_MESSAGE', 'unknown')[/[^\n]+/]"
+        $abi, $plat = $(ruby -e "STDOUT.write RbConfig::CONFIG['ruby_version'] + ' ' + RUBY_PLATFORM").split(' ')
+        if (($abi -ge '2.4') -and $plat.Contains('mingw')) {
+          ridk version
+        } else {
+          echo 'ridk is unavailable'
+        }
     - name: Subprocess test
       run: ruby test_subprocess.rb
     - name: OpenSSL version
       run: ruby -ropenssl -e 'puts OpenSSL::OPENSSL_LIBRARY_VERSION'
     - name: OpenSSL test
       run: ruby -ropen-uri -e 'puts open(%{https://rubygems.org/}) { |f| f.read(1024) }'
     - name: C extension test
+      if: endsWith(matrix.ruby, 'mswin') == false
       run: gem install json:2.2.0 --no-document
+    - name: C extension test (mswin)
+      shell: cmd
+      if: endsWith(matrix.ruby, 'mswin')
+      run: |
+        call "C:\Program Files (x86)\Microsoft Visual Studio\${{ matrix.vs }}\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```

Comment:
It seems quite inconvenient to have to use this long line before installing any C extension.
Is this something the action could setup once and for all? Or it has to be done in each shell session?

I think very few Rubyists expect having to anything extra before being able to install C extensions.

Reply:
Again, this is mess.  I've defined the string as an ENV variable in actions-ruby.

The `vcvars*.bat` files are needed when using msvc from the command line, as the scripts set up the environment.  So, if one needs the tools in multiple steps (which isolate the ENV from user script changes), one needs to run the script in each one until there's a better solution.  I am looking into one.

Some of this is made messier by Actions image options.  I haven't checked recently, but they announced that they were removing the Window-2016 image.  They are still updating the code for it.

So, although they've stated that they'll only have one Windows image, that will likely change when they release another version of Windows Server.  That's the image side, then there is the Visual Studio ('VS') side.  There's only one version of VS in each image, but that may also change.  So, again, I may see if I can get answers about the singular/multiple issues, as that effects the code and needed inputs in custom actions.

Reply:
If https://github.com/MSP-Greg/actions-ruby/blob/164cd7048c050264a0360510c572e1d4472b4bfb/index.js#L139 is enough to let `gem install json` work with mswin builds, I think we should do that in this action too.

Reply:
Ok.  I'll set it up assuming 'singular' (as does actions-ruby) and hope we can keep an eye on Actions images.

Reply:
Done.  The Actions job runs it both in a cmd shell and the default PowerShell shell.  A little different.  I've got a bunch of issues to open re the Actions runner, one will be about adding the setting in VCVARS to the env with an action...